cmd/ci-operator/main.go

```diff
@@ -743,6 +750,18 @@ func (o *options) writeMetadataJSON() error {
 		m.Repo = fmt.Sprintf("%s/%s", o.jobSpec.Refs.Org, o.jobSpec.Refs.Repo)
 		m.Repos = map[string]string{m.Repo: o.jobSpec.Refs.String()}
 	}
+	if len(o.jobSpec.ExtraRefs) > 0 {
+		if m.Repos == nil {
```

Comment:
Why only in this case? It's possible to have refs and extra refs

Reply:
we don't need to set repos non-nil if we aren't adding anything to o it.

Reply:
aren't you also doing the insertion after that conditionally, though?